abahachuk/ptushki-backend
# update: EPMCUXDA-7676 initial data shape

### What does this PR do?
This PR changes the format of the data to be returned: instead of returning a field related to the selected language, e.g. `desc_byn` -- will always return a common `desc` field.

This PR also makes a number of improvements:

1. The caching policy has been redefined -- now we have two types of caches, one common record + related and extracted entity method, now it's used to heat the cache, and the second one for the endpoint `initial-data` with filtration by language, in principle it allows not to filter each time that increases the throughput from 250ms to 80ms at local testing.
2. The endpoint contract was also explicitly declared, 
3. and the filtration process was slightly redesigned.
4. `EntityDto` was renamed to `EURINGEntityDto` as too common name



### How should I test this?

Try to get initial data by querying corresponding endpoint with next queries:

1. `{{ptushki_url}}/initial-data`
2. `{{ptushki_url}}/initial-data?lang=ru` 
3. `{{ptushki_url}}/initial-data?lang=byn` 
4. `{{ptushki_url}}/initial-data?lang=rus` 
5. `{{ptushki_url}}/initial-data?lang=eng` 
6. `{{ptushki_url}}/initial-data?lang=byn` 

Essentially, you need to confirm that if the language is incorrect or absent, data for English is sent, and if the language is supported, data for the selected language.

**Supported languages:** `eng`, `rus`, `byn`

**Please note** that the fixtures are not perfect, and for Russian and Belarusian you will mostly receive English descriptions.

### Jira ticket related to this PR

- [EPMCUXDA-7676](https://jira.epam.com/jira/browse/EPMCUXDA-7676)

### Type of change

- [ ] Adds new features.
- [x] Changes existing functionality.
- [ ] Deprecates features.
- [ ] Removes features.
- [ ] Fixes any bug.
- [ ] Other: security, build process, infrastructure, tests, docs, etc.
